roles/splunk_common/tasks/get_spark.yml

```diff
@@ -0,0 +1,39 @@
+---
+- name: Define versioned file name
+  set_fact:
+    versioned_name: "spark-{{ splunk.dfs.spark_version }}-bin-hadoop{{ splunk.dfs.hadoop_version }}"
+
+- name: Download Hadoop
+  include_tasks: s3_download.yml
+  vars:
+    file_name: "{{ versioned_name }}.tgz"
+
+- name: Untar Spark
+  unarchive:
+    src: "/tmp/{{ versioned_name }}.tgz"
+    dest: "/tmp/"
+    copy: no
+
+- name: Create Spark directory
+  file:
+    path: "{{ splunk.dfs.spark_home }}"
+    mode: 0777
+    state: directory
+  become_user: "{{ privileged_user }}"
+  become: yes
+  register: spark_directory
+
+- name: Move Spark package to directory
+  shell: /bin/bash -c "mv /tmp/{{ versioned_name }}/* {{ splunk.dfs.spark_home }}"
+  when:
+    - spark_directory is defined
+
+- name: Chmod Spark directory
```

Comment:
I'm not sure if you need this task either - this sets the entire `spark_home` dir to be owned by the `splunk:splunk` user/group, but when you create the configurations and start `spark`, it is being run as `root`.

Should we be running `spark` as root? Or do you want to run it as the `splunk` user?

Reply:
We didn't see any issues starting spark as the root user, however the spark logs complained about access to files under the /opt/spark directory once the DFS search is executed. DFS team explained that they expected the directory/files to be owned by the same user that is starting splunk, but I'm still not sure why...
I can test starting spark as the splunk user to be consistent, but let's discuss this further.